### PR TITLE
Removing EfficientNet model head

### PR DESCRIPTION
**Pull request by S-Lykles**
The EfficientNetEncoders (both the timm and the normal one) have a final expanding 1x1 convolutional layer followed by a batchnorm layer which are not being used in the forward call. I suggest deleting these layers similar to how the fully connected layer is deleted. Deleting these layers saves on some model parameters. 
I came across this when I was training a model in ddp using pytorch-lightning, I was getting an error because some of the model parameters were not being used for calculating the loss and therefore the gradients remain "None" whilst ddp is still trying to sync all the gradients between devices. There is a setting to deal with unused parameters but it makes the training slightly slower.

### CODE REVIEW

**Comment by S-Lykles** [comment]
Found out this would break loading pre-trained weights, added a few lines to "load_state_dict" to fix this.
